R/capture-requests.R

```diff
@@ -429,3 +434,41 @@ get_redactor <- function() {
 
   return(NULL)
 }
+
+#' Check dittodb environment path
+#'
+#' This function should generally not be used, but must be exported for the
+#' query recording function to work properly
+#'
+#' @param .dittodb_env Environment object
+#' @param n Integer. Type of error to throw.
+#'
+#' @return NULL, invisibly.
+#' @export
+check_db_path <- function(.dittodb_env, n = 1L) {
+  if(is.null(.dittodb_env$db_path)) {
+    if (n == 1L) {
+      msg <- c("Database capture failed",
+               "*" = "The database connection object was created before calling 'start_db_capturing()'",
+               "*" = "Please close the connection and ensure it's created after calling 'start_db_capturing()'.")
+
+      trc <- NULL
+    } else {
+      msg <- ""
+
+      trc <- structure(
+        .Data = "",
+        class = c("rlang_trace", "rlib_trace", "tbl", "data.frame")
+      )
+    }
```

Comment:
Ah, I think I led you slightly astray here. I forgot that because of the way that these recordings happen, we will actually hit this twice, and I didn't totally recognize that when I commented before. 

I see here you have the second error (or first, depending on one's perspective of the callstack!) have no message. 

I know that it's _possible_ with rlang to trigger these conditions and have them roll up and produce only one, but honestly I think that's probably too much work + extra code for this feature. Really here we don't mind if this is super noisy with too many errors all at once if someone calls this. It's not the most elegant but it'll get the message across.

So what about: we always produce the verbose message (so basically, sorry the tabs are off here, I'm typing it out in the comment field):

```
check_db_path <- function(.dittodb_env, n = 1L) {
  if(is.null(.dittodb_env$db_path)) {
        rlang::abort(
      message = c("Database capture failed",
               "*" = "The database connection object was created before calling 'start_db_capturing()'",
               "*" = "Please close the connection and ensure it's created after calling 'start_db_capturing()'."),
      call = rlang::caller_env()
    )
}

  return(invisible())
}
``` 

And the output would end up being like:

```
> dbListTables(con)
Error in `dbSendQuery()`:
! Database capture failed
• The database connection object was created before calling 'start_db_capturing()'
• Please close the connection and ensure it's created after calling 'start_db_capturing()'.
Run `rlang::last_trace()` to see where the error occurred.
Error in `dbListTables()`:
! Database capture failed
• The database connection object was created before calling 'start_db_capturing()'
• Please close the connection and ensure it's created after calling 'start_db_capturing()'.
Run `rlang::last_trace()` to see where the error occurred.
```

It's a little over-verbose but that's ok — and simpler for us to implement!

Reply:
updated